ImputationPipeline/Imputation.wdl

```diff
@@ -0,0 +1,475 @@
+  File to_be_imputed_vcf_index = select_first([QConArray.output_vcf_index, GenerateDataset.output_vcf_index])
+
+  scatter (chrom in ["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22"]) { 
+    call CalculateChromsomeLength {
+      input:
+        ref_dict = ref_dict,
+        chrom = chrom
+    }
+    
+    Float chunkLengthFloat = chunkLength
+    Int num_chunks = ceil(CalculateChromsomeLength.chrom_length / chunkLengthFloat)
+
+    scatter (i in range(num_chunks)) {
+      call ChunkVCF {
+        input:
+          start = (i * chunkLength) + 1,
+          end = if (CalculateChromsomeLength.chrom_length < ((i + 1) * chunkLength)) then CalculateChromsomeLength.chrom_length else ((i + 1) * chunkLength), 
+          vcf = to_be_imputed_vcf,
+          vcf_index = to_be_imputed_vcf_index,
+          chrom = chrom,
+          basename = "chrom_" + chrom + "_chunk_" + i
+       }
+
+      call QConChunk {
+        input: 
+          vcf = ChunkVCF.output_vcf,
+          vcf_index = ChunkVCF.output_vcf_index,
+          panel_bcf = path_to_reference_panel + "ALL.chr" + chrom + ".phase3_integrated.20130502.genotypes.bcf",
+          panel_bcf_index = path_to_reference_panel + "ALL.chr" + chrom + ".phase3_integrated.20130502.genotypes.bcf.csi",
+          sarah_jar = new_gatk_jar
+      }
+
+      call PrePhaseVariantsEagle {
+        input:
+          dataset_bcf = QConChunk.valid_chunk_bcf,
+          dataset_bcf_index = QConChunk.valid_chunk_bcf_index,
+          reference_panel_bcf = path_to_reference_panel + "ALL.chr" + chrom + ".phase3_integrated.20130502.genotypes.bcf",
+          reference_panel_bcf_index = path_to_reference_panel + "ALL.chr" + chrom + ".phase3_integrated.20130502.genotypes.bcf.csi",
+          chrom = chrom,
+          valid_chunk = QConChunk.valid,
+          genetic_map_file = genetic_maps_eagle,
+          start = (i * chunkLength) + 1,
+          end = (i + 1) * chunkLength + 5000000 # they do an overlap of 5,000,000 bases in michigan pipeline
+      }
+
+        call minimac4 {
+          input:
+            ref_panel = path_to_m3vcf + chrom + ".1000g.Phase3.v5.With.Parameter.Estimates.m3vcf.gz",
+            phased_vcf = PrePhaseVariantsEagle.dataset_prephased_vcf,
+            prefix = "chrom" + "_chunk_" + i +"_imputed",
+            chrom = chrom,
+            valid_phased_vcf = QConChunk.valid,
+            start = (i * chunkLength) + 1,
+            end = (i + 1) * chunkLength
+        }
+     }
+  }
+
+  Array[File] phased_vcfs = flatten(minimac4.vcf)
+
+  call MergeVCFs {
+    input:
+      input_vcfs = phased_vcfs,
+      output_vcf_basename = output_callset_name,
+      shard_size = size(minimac4.vcf[0], "GB")
+  }
+
+  File michiganserver_vcf = MergeVCFs.output_vcf
+
+  call UpdateHeader {
+    input:
+      vcf = michiganserver_vcf,
+      ref_dict = ref_dict
+  }
+
+  call RemoveSymbolicAlleles as RSAlleles {
```

Comment:
why the special alias?

Reply:
I think Edyta was using the job twice, this is unnecessary now, thanks for catching 👍